drainer/collector.go

```diff
@@ -344,11 +344,24 @@ func (c *Collector) keepUpdatingStatus(ctx context.Context, fUpdate func(context
 		case <-ctx.Done():
 			return
 		case nr := <-c.notifyChan:
-			nr.err = fUpdate(ctx)
+			// stop merge and return directly
+			// temporal logic:
+			// * merge is paused -> new pump write binlog
+			// * fetch all sources in merge -> find smallest commit ts binlog -> check merge whether paused -> publish binlog
+			// so it's safe
+			// TODO: maybe we can remove merge.SourceChanged
+			c.merger.Stop()
 			nr.wg.Done()
+			if err := fUpdate(ctx); err != nil {
+				log.Error("Failed to update collector status (notifying drainer progress)", zap.Error(err))
```

Comment:
maybe we can add a failpoint test here later

Reply:
tidb binlog doesn't have failpoint

Reply:
We don't need to use `failpoint` here because `fUpdate` is a parameter you can pass. To test error handling we can just pass in a function that returns error as we like.

Reply:
there're not any error handling here now, we need to test the temporal logic, it needs failpoint